node/src/components/consensus.rs

```diff
@@ -103,14 +105,23 @@ pub enum Event<I> {
         delay: Duration,
     },
     /// Event raised when a new era should be created: once we get the set of validators, the
-    /// booking block hash and the seed from the key block
+    /// booking block hash and the seed from the key block.
     CreateNewEra {
         /// The header of the switch block
         block_header: Box<BlockHeader>,
         /// Ok(block_hash) if the booking block was found, Err(height) if not
         booking_block_hash: Result<BlockHash, u64>,
     },
-    /// An event instructing us to shutdown if the latest era received no votes
+    /// Event raised upon initialization, when a number of eras have to be instantiated at once.
+    InitializeEras {
+        switch_blocks: HashMap<EraId, BlockHeader>,
+        current_era: EraId,
+        validators: BTreeMap<PublicKey, U512>,
+        genesis_state_root_hash: Digest,
```

Comment:
Shouldn't the era supervisor should be getting this from the chainspec?

Reply:
I think this is being passed from the chainspec. But it shouldn't really be `genesis_*`, as Andreas pointed out above. I'm not sure how to get the values for the right block right now, though - I'm actually leaning towards marking it as a TODO for when fast sync and code related to emergency upgrades land...